src/builtin_string.cpp

```diff
@@ -949,6 +949,88 @@ static int string_split(parser_t &parser, io_streams_t &streams, int argc, wchar
     return splits.size() > arg_count ? BUILTIN_STRING_OK : BUILTIN_STRING_NONE;
 }
 
+static int string_repeat(parser_t &parser, io_streams_t &streams, int argc, wchar_t **argv) {
+	const wchar_t *short_options = L":t:q";
+	const struct woption long_options[] = {{L"times", required_argument, 0, 't'},
+										   {L"quiet", no_argument, 0, 'q'},
+										   {0, 0, 0, 0}};
+
+	bool quiet = false;
+	long times = 0;
+	wgetopter_t w;
+
+	for (;;) {
+		int opt;
+
+		if ((opt = w.wgetopt_long(argc, argv, short_options, long_options, 0)) == -1) {
+			break;
+		}
+
+		switch(opt) {
+		case 0: {
```

Comment:
Since it's my first time contributing to fish, i wanted to stay as close style-wise to other functions. I will rewrite this in a more correct way.